source/common/upstream/health_checker_base_impl.cc

```diff
@@ -419,6 +419,31 @@ HealthCheckerImplBase::ActiveHealthCheckSession::clearPendingFlag(HealthTransiti
 }
 
 void HealthCheckerImplBase::ActiveHealthCheckSession::onIntervalBase() {
+  // Only if host is healthy, check whether need disable HC if there is already business traffic
+  if (parent_.disable_health_check_if_active_traffic &&
+      !host_->healthFlagGet(Host::HealthFlag::FAILED_ACTIVE_HC)) {
+    MonotonicTime traffic_pass_time;
+    switch (parent_.healthCheckerType()) {
+    case envoy::data::core::v3::HealthCheckerType::TCP:
+      traffic_pass_time = host_->lastTrafficPassTime();
+      break;
+    case envoy::data::core::v3::HealthCheckerType::HTTP:
+      traffic_pass_time = host_->lastTrafficPassTime2xx();
+      break;
+    case envoy::data::core::v3::HealthCheckerType::GRPC:
+      traffic_pass_time = host_->lastTrafficPassTimeGrpc();
+      break;
+    default:
+      break;
+    }
+    auto duration = time_source_.monotonicTime() - traffic_pass_time;
+    if (duration.count() < std::chrono::milliseconds(parent_.interval_ * 1000000).count()) {
```

Comment:
I'm not sure why we are multiplying by 1,000,000 here. Can you use duration casts and direct duration comparison to make this more clear?

Reply:
fixed